src/libaktualizr/crypto/crypto.h

```diff
@@ -58,10 +58,50 @@ class PublicKey {
   KeyType type_{KeyType::kUnknown};
 };
 
+namespace Uptane {
+/**
+ * The hash of a file or TUF metadata.  File hashes/checksums in TUF include the length of the object, in order to
+ * defeat infinite download attacks.
+ */
+class Hash {
+ public:
+  // order corresponds algorithm priority
+  enum class Type { kSha256, kSha512, kUnknownAlgorithm };
+
+  static Hash generate(Type type, const std::string &data);
+  Hash(const std::string &type, const std::string &hash);
+  Hash(Type type, const std::string &hash);
```

Comment:
Does it make sense to delete the default constructor? Otherwise, it will initialize `type_` to `kSha256`, which is probably not what you want.
I'm also not sure what is the purpose of `kUnknownAlgorithm` objects.

Reply:
Sorry, I just noticed that you've actually simply moved this code.

Reply:
Why was it necessary to move this, though? It's weird to define `Uptane` namespace objects outside of the uptane subdirectory. I'd like to avoid that if possible. This might be a refactoring that we can pull out and merge immediately even if we can't merge the rest yet.

Reply:
But there is also nothing Uptane specific here, it's just a generic crypto helper. I'd probably move it under the `Crypto` namespace, which can replace the `Crypto` class below.

Reply:
So, I just move the Hash out of the Uptane namespace and leave it in crypto.h/cc?

Reply:
> So, I just move the Hash out of the Uptane namespace and leave it in crypto.h/cc?

That sounds reasonable to me.